charts/karpenter/crds/karpenter.sh_provisioners.yaml

```diff
@@ -41,6 +41,68 @@ spec:
               Node properties are determined from a combination of provisioner and
               pod scheduling constraints.
             properties:
+              instanceTypeFilter:
+                description: InstanceTypeFilter allows filtering the instance types
+                  provided by the cloud provider.
+                properties:
+                  cpuCount:
+                    description: CPUCount allows filtering instance types by the min
+                      and maximum number of CPUs on the instance type with the minimum
+                      and maximum values being inclusive.
+                    properties:
+                      max:
+                        description: Max is the minimum value for the filter
+                        format: int64
+                        minimum: 0
+                        type: integer
+                      min:
+                        description: Min is the minimum value for the filter.
+                        format: int64
+                        minimum: 0
+                        type: integer
+                    type: object
+                  memoryMiB:
+                    description: MemoryMiB allows filtering instance types by the
+                      min and maximum MiB of memory on the instance type with the
+                      minimum and maximum values being inclusive.
+                    properties:
+                      max:
+                        description: Max is the minimum value for the filter
+                        format: int64
+                        minimum: 0
+                        type: integer
```

Comment:
why aren't these resource quantitys?

Reply:
That makes sense, I was modeling after the ABS but what if we dropped CPU/Memory altogether and made it generic with just min & max resource lists?

Something like:
```yaml
instanceTypeFilter:
  minResources:
    cpu: 2
    memory: 8Gib
  maxResources:
    cpu: 8
    nvidia.com/gpu: 0
```

Reply:
hmm that is kind of nice... It seems like it's getting a little into the requirements territory though. Wouldn't you want to set a `DoesNotExist` (or `LT`) on `nvidia.com/gpu` in requirements to filter more generically?  Requirements could also support special capabilities like networking or NVMe via labels that the CP adds to nodes.

Reply:
That was probably a bad example. If you wanted only instance types with 2 or more GPUs, you could filter generically using resources.  We can't support that currently unless you list instance types.  I'm not 100% sold on it, but the thought of going to quantities made me think of just generic resource list min/max, which eliminates the need to revisit this if/when someone wants to filter based on some other resource type.

Reply:
I like the semantic of it. It's growing on me :D